addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/internal/discovery/LightifyDeviceDiscoveryService.java

```diff
@@ -0,0 +1,107 @@
+package org.openhab.binding.lightify.internal.discovery;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.lightify.handler.DeviceHandler;
+import org.openhab.binding.lightify.handler.GatewayHandler;
+import org.openhab.binding.lightify.internal.link.LightifyLight;
+import org.openhab.binding.lightify.internal.link.LightifyLink;
+import org.openhab.binding.lightify.internal.link.LightifyLuminary;
+import org.openhab.binding.lightify.internal.link.LightifyZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_DEVICE_ADDRESS;
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_DEVICE_NAME;
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ID;
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ZONE_ID;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_BULB;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_ZONE;
+import static org.openhab.binding.lightify.internal.LightifyUtils.exceptional;
+
+public class LightifyDeviceDiscoveryService extends AbstractDiscoveryService implements Consumer<LightifyLuminary> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LightifyDeviceDiscoveryService.class);
+
+    private final static int SEARCH_TIME = 60;
+
+    private final GatewayHandler gatewayHandler;
+
+    public LightifyDeviceDiscoveryService(GatewayHandler gatewayHandler) throws IllegalArgumentException {
+        super(DeviceHandler.SUPPORTED_TYPES, SEARCH_TIME, false);
+        this.gatewayHandler = gatewayHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        exceptional(() -> {
+            LOGGER.info("Start scanning for paired devices");
+            LightifyLink lightifyLink = gatewayHandler.getLightifyLink();
+            lightifyLink.performSearch(this);
+        });
+    }
+
+    @Override
+    public void accept(LightifyLuminary luminary) {
+        exceptional(() -> {
+            LOGGER.info("Found device: " + luminary);
```

Comment:
2. Pelase change logging level to `debug`
3. Please use parametrized logging instead of string concatenation.